### PR TITLE
small type cleanup: remove exclamation points

### PR DESCRIPTION
**Pull request by samouri**
**summary**
In closure, most types are nullable by default and required `!` to mark them as NonNullable.
This is no longer necessary in TS. This PR applies @rcebulko 🧙 magic🧙   SublimeText regex to a couple of directories to remove the needless noise from future PRs. 

**regex**:
- Find `((?:^ *|\/\*)\* @\w+ (?:[^*\n]|\*(?!\/))*)!(.*?(?:\*\/|$))`
- Replace: `$1$2`

**directories**
- extensions/amp-carousel/0.1
- src/preact
- src/core

### CODE REVIEW

File: extensions/amp-carousel/0.1/slidescroll.js [hunk 1]
```diff
@@ -67,7 +67,7 @@ export class AmpSlideScroll extends AMP.BaseElement {
     /** @private {?Element} */
     this.slidesContainer_ = null;
 
-    /** @private {!Array<!Element>} */
+    /** @private {!Array<Element>} */
```
[the remark this answers is not in the document]
Comment (rcebulko) [reply extensions/amp-carousel/0.1/slidescroll.js:70]: This only happens in the case of function/object/generic types like this one, and I scanned the PR and found this is the only straggler.
```suggestion
    /** @private {Array<Element>} */
```

**Review by rcebulko — APPROVED** [review]
pending one suggestion

File: extensions/amp-carousel/0.1/slidescroll.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (samouri) [reply extensions/amp-carousel/0.1/slidescroll.js]: Ooh, found 5 stragglers by rerunning. Thanks!

**Review by jridgewell — APPROVED** [review]

**Comment by jridgewell** [comment]
You can also extend `build-system/eslint-rules/closure-type-primitives.js` to remove from these files.

**Pull request merged**